src/Tasks/MigrateContentToElement.php

```diff
@@ -17,26 +55,121 @@ class MigrateContentToElement extends BuildTask
 
     public function run($request)
     {
-        // TODO: needs rewriting for multiple elemental areas
         $pageTypes = singleton(ElementalArea::class)->supportedPageTypes();
         $count = 0;
         foreach ($pageTypes as $pageType) {
-            $pages = $pageType::get()->filter('ElementalAreaID', 0);
+            // Only pages that have the ElementalPageExtension have a known ElementalArea relation
+            if (!$this->isMigratable($pageType)) {
+                continue;
+            }
+
+            $pages = $pageType::get()->filter('Content:not', ['', null]);
+            $clearContent = $this->config()->get('clear_content');
+
+            $this->extend('updatePageFilter', $pages, $pageType);
+            $pageTypeCount = 0;
+
+            /** @var SiteTree&ElementalAreasExtension $page */
             foreach ($pages as $page) {
+                if ($this->shouldSkipMigration($page)) {
+                    continue;
+                }
+                // Fetch and clear existing content (if configured)
                 $content = $page->Content;
-                $page->Content = '';
-                // trigger area relations to be setup
-                $page->write();
-                $area = $page->ElementalArea();
-                $element = new ElementContent();
+                $pageIsLive = $page->isPublished();
+                if ($clearContent) {
+                    $page->Content = '';
+                }
+
+                // Get the area
+                $area = $this->getAreaRelationFromPage($page);
+
+                // Write the page if we're clearing content or if the area doesn't exist - we write to trigger a
+                // relationship update
+                if ($clearContent || !$area->exists()) {
+                    try {
+                        $page->write();
```

Comment:
I guess I was more meaning a kind of transaction style operation where migrating the content into an element fails, you roll it back so you don't clear the page content

Reply:
Oh yep I understand - I'll follow that up later

Reply:
I think I'll merge this for now, if you want to do another patch change to add transactional style rollbacks that'd be cool but not blocking merge